symengine/tests/basic/test_basic.cpp

```diff
@@ -383,6 +383,13 @@ TEST_CASE("compare: Basic", "[basic]")
     r2 = mul(x, z);
     CHECK(r1->compare(*r2) == -1);
     CHECK(r2->compare(*r1) == 1);
+//    CHECK(r1->compare(*r2) == -1);
+//    CHECK(r2->compare(*r1) == 1);
+
+    r1 = mul(y, x);
+    r2 = mul(x, z);
+//    CHECK(r1->compare(*r2) == -1);
+//    CHECK(r2->compare(*r1) == 1);
```

Comment:
I see. We might need two implementations of `__cmp__`, one fast but platform dependent, and another slower but platform independent (that one will be used in printing).